catt/cli.py

```diff
@@ -218,25 +217,22 @@ def cast(settings, video_url, subtitle, force_default, random_play, no_subs):
             cst.kill(idle_only=True)
             raise CattCliError("Playlist is empty.")
         click.echo("Casting remote playlist %s..." % video_url)
-        if random_play:
-            stream.set_playlist_entry(random.randrange(0, stream.playlist_length))
+        if not random_play and cst.playlist_capability and stream.playlist_all_ids:
+            cst.play_playlist(stream.playlist_all_ids)
         else:
-            try:
-                if not stream.playlist_all_ids:
-                    raise ValueError
-                cst.play_playlist(stream.playlist_all_ids)
-                return
-            except (PlaybackError, ValueError):
+            if random_play:
+                stream.set_playlist_entry(random.randrange(0, stream.playlist_length))
```

Comment:
Do you want to split the call to `set_playlist_entry()` here outside the `if` so it's not repeated?

Reply:
I'm not sure I understand what you mean.

Reply:
Just:

```python
if random_play:
    entry = random.randrange(0, stream.playlist_length)
else:
    warning("Playlist playback not possible, playing first video.")
    entry = 0
stream.set_playlist_entry(entry)
```

Reply:
Ah, ok. Hang on.

Reply:
I'm watching formula 1, so I'm not entirely focused. :smile:

Reply:
Haha, understandable :P